store/store.go

```diff
@@ -1255,9 +1270,50 @@ func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *sna
 		return "", err
 	}
 
+	// TODO: Check sha3_384
 	return deltaPath, nil
 }
 
+// applyDelta generates a target snap from the current snap and a downloaded delta.
+var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo) (string, error) {
```

Comment:
This is a variable (in the same way the private download function is) so that the public Download function can be tested without being requiring the setup necessary for applyDelta, while applyDelta can itself be unit-tested using the MockCommand as below.

If it's better to just do the extra setup in the Download unit-tests, let me know and I'll update to that.